File: QuizApp/app/src/main/java/com/example/quizapp/MainActivity.java
package com.example.quizapp;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {
    EditText Username;

    public void next(View view) {
        Intent intent = new Intent(this,Question1.class);

        intent.putExtra("username",Username.getText().toString());


        startActivity(intent);
    }


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

         Username = findViewById(R.id.username);


    }



}

File: androidapp/app/src/main/java/com/kajan/restapi/MainActivity.kt
package com.kajan.restapi

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.github.kittinunf.result.Result
import com.github.kittinunf.fuel.*
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        button.setOnClickListener {

            val httpAsync = "https://jsonplaceholder.typicode.com/posts"
                .httpGet()
                .responseString { request, response, result ->
                    when (result) {
                        is Result.Failure -> {
                            val ex = result.getException()
                            println(ex)
                        }
                        is Result.Success -> {
                            val data = result.get()
                            println(data)
                            
                        }
                    }
                }

            httpAsync.join()
        }
    }
}
